music21/articulations.py

```diff
@@ -582,10 +583,27 @@ class PullOff(spanner.Spanner, TechnicalIndication):
     pass
 
 class FretBend(FretIndication):
-    bendAlter: interval.IntervalBase | None = None
-    preBend: t.Any = None
-    release: t.Any = None
-    withBar: t.Any = None
+    bendAlter: interval.IntervalBase
+    preBend: bool = False
+    release: t.Optional[float]
+    withBar: t.Optional[bool]
+
+    def __init__(self, number=0, preBend=False, release=None, withBar=None, **keywords):
```

Comment:
number is too generic a term -- and we usually reserve the term for identifying which of multiple simultaneous items (like slur number=1 vs slur number=2).   Are fret-bends always chromatic intervals?  it seems like you can also have a diatonic interval of a fretbend.  hence why bendAlter was typed as IntervalBase not ChromaticInterval.  It seems better to have bendAlter as described above `interval.IntervalBase | None = None` where None indicates unspecified.  Having a default of 0, becoming ChromaticInterval(0) indicates that this fretBend does not alter the pitch at all, which would be a very strange fret bend.  This would also make it so that interval.py does not need to be imported here.

The other keywords should be typed in the `__init__` declaration in addition to the class.

We no longer use `t.Optional[X]` in the code base.  Use `X | None` instead.

Reply:
Changed with https://github.com/cuthbertLab/music21/pull/1580/commits/d63c78ab54208bc0d8da8558190b821329900809